### PR TITLE
Acquire language server binaries during build

### PR DESCRIPTION
**Pull request by StephenWeatherford**
If environment variables are set with credentials to the language server nuget package, then they are downloaded and incorporated into the vsix during build, and the license is swapped out for the one with the language server.
Otherwise, a vsix without the language server is created.

A staging directory is used so we can modify the package/license as needed.

Once this is merged, I can merge saw/language-server into master.

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
Alex/Prashanth: I'd like to get this in today, thanks!

File: gulpfile.ts [hunk 1]
```diff
@@ -97,45 +128,181 @@ function buildTLEGrammar(): void {
...
-    // let armServerBin = path.join(env.ExtensionsBin, '..', '..', 'ARM-LanguageServer', 'Microsoft.ArmLanguageServer', 'bin', 'Debug', 'netcoreapp2.2', 'publish');
-    let armServerBin = '\\\\scratch2\\scratch\\stephwe\\ARM\\Dev Assemblies\\Current';
-    let updateDest = path.join(__dirname, languageServerFolderName);
-    if (!fs.existsSync(updateDest)) {
-        fs.mkdirSync(updateDest);
+function executeInShell(command: string): void {
+    console.log(command);
+    const result = shelljs.exec(command);
+    console.log(result.stdout);
+    console.log(result.stderr);
```
Comment (PrashanthCorp) [inline gulpfile.ts]: Is this writing to the terminal? Is there a reason we don't use ext's `outputChannel` here?

Comment (StephenWeatherford) [reply gulpfile.ts]: This is gulp, thus during build.  No output channel.

File: gulpfile.ts [hunk 2]
```diff
@@ -97,45 +128,181 @@ function buildTLEGrammar(): void {
...
 async function buildGrammars(): Promise<void> {
-    if (!fs.existsSync('dist')) {
-        fs.mkdirSync('dist');
+    if (!fse.existsSync('dist')) {
+        fse.mkdirSync('dist');
     }
-    if (!fs.existsSync('dist/grammars')) {
-        fs.mkdirSync('dist/grammars');
+    if (!fse.existsSync('dist/grammars')) {
+        fse.mkdirSync('dist/grammars');
```
[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply gulpfile.ts]: No great reason, but it's simpler and there's no need for parallelism during this part of build.

[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply gulpfile.ts]: Thanks for the tip, fixed.

**Pull request merged**